src/backend/cuda/blas.cpp

```diff
@@ -229,7 +236,7 @@ Array<T> matmul(const Array<T> &lhs, const Array<T> &rhs, af_mat_prop optLhs,
                 getType<T>(), lStrides[1], rhs.get(), getType<T>(), rStrides[1],
                 &beta, out.get(), getType<T>(), out.strides()[1],
                 getType<T>(),  // Compute type
-                CUBLAS_GEMM_DEFAULT_TENSOR_OP));
+                selectGEMMAlgorithm()));
```

Comment:
This change should actually be a separate commit by itself when we merge this PR. Leaving a comment here so that we don't forget to reorganize the commits.

Reply:
We aren't using that function before this change. Additionally that function is not compatible with Kepler. Need to find another way to call it.

Reply:
Nevertheless, it is not fp16 related change. Hence, my comment.